packages/gatsby/src/services/initialize.ts

```diff
@@ -32,6 +32,31 @@ interface IPluginResolution {
   options: IPluginInfoOptions
 }
 
+// If the env variable GATSBY_EXPERIMENTAL_FAST_DEV is set, enable
+// all DEV experimental changes (but only during development & not on CI).
+if (
+  process.env.gatsby_executing_command === `develop` &&
+  process.env.GATSBY_EXPERIMENTAL_FAST_DEV &&
+  !isCI()
+) {
+  process.env = {
+    ...process.env,
+    GATSBY_EXPERIMENTAL_LAZY_DEVJS: `true`,
+    GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND: `true`,
+    GATSBY_EXPERIMENTAL_DEV_SSR: `true`,
+  }
```

Comment:
I don't think overwriting entire `process.env` is safe

it changes the way `process.env.` setter works

for example:

```
process.env.WAT = 5

console.log({ test: process.env.WAT })

process.env = {
  ...process.env,
  WAT: 1,
}

console.log({ test: process.env.WAT })
```

will result in

```
{ test: '5' }
{ test: 1 }
```

So overwrite removes Node's default to always cast to strings

Reply:
( this might have unforseen consequences both to our code and dependencies we use )

Reply:
Yeah, I think https://github.com/gatsbyjs/gatsby/pull/28166/commits/f07e9cc1cc98b825989fe2486371b1ee6533de43 is good 👍 

Not as sleek looking, but safer